### PR TITLE
Fix delete functionality to support "Select Action" dropdown

### LINKED ISSUE #4723: Fix delete operations for all katello entities
Connected to #4721

As all corresponding buttons were removed from UI, we have to fix our code to support new approach

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #4723
Part of #4721
Part of #3101

That PR should fix approximately 30 test cases. That is a first phase of moving to singular delete method across the code

```
nosetests tests/foreman/ui/test_activationkey.py -m _delete
..E..E
----------------------------------------------------------------------
Ran 6 tests in 1503.719s

FAILED (errors=2)


nosetests tests/foreman/ui/test_architecture.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 92.516s

OK

nosetests tests/foreman/ui/test_computeprofile.py -m _delete
.
----------------------------------------------------------------------
Ran 1 test in 123.809s

OK

nosetests tests/foreman/ui/test_computeresource.py -m _delete
.
----------------------------------------------------------------------
Ran 1 test in 187.643s

OK

nosetests tests/foreman/ui/test_config_group.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 110.849s

OK

nosetests tests/foreman/ui/test_contentview.py -m test_positive_delete
..S...E
----------------------------------------------------------------------
Ran 7 tests in 990.284s

FAILED (SKIP=1, errors=1)

nosetests tests/foreman/ui/test_domain.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 40.166s

OK

nosetests tests/foreman/ui/test_environment.py 
....
----------------------------------------------------------------------
Ran 4 tests in 289.853s

OK

nosetests tests/foreman/ui/test_gpgkey.py -m _delete
...EE..ES
----------------------------------------------------------------------
Ran 9 tests in 1150.093s

FAILED (SKIP=1, errors=3)

nosetests tests/foreman/ui/test_host.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 80.542s

OK

nosetests tests/foreman/ui/test_hostcollection.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 276.373s

OK


nosetests tests/foreman/ui/test_hostgroup.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 151.965s

OK

nosetests tests/foreman/ui/test_hw_model.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 103.885s

OK


nosetests tests/foreman/ui/test_lifecycleenvironment.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 58.726s

OK

nosetests tests/foreman/ui/test_location.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 137.656s

OK

nosetests tests/foreman/ui/test_medium.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 41.615s

OK


nosetests tests/foreman/ui/test_operatingsystem.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 39.259s

OK

nosetests tests/foreman/ui/test_organization.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 196.059s

OK

nosetests tests/foreman/ui/test_partitiontable.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 123.943s

OK

nosetests tests/foreman/ui/test_product.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 243.731s

OK

nosetests tests/foreman/ui/test_puppetclass.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 120.423s

OK

nosetests tests/foreman/ui/test_role.py -m test_positive_delete
.E
----------------------------------------------------------------------
Ran 2 tests in 195.492s

FAILED (errors=1)

nosetests tests/foreman/ui/test_subnet.py -m _delete
..
----------------------------------------------------------------------
Ran 2 tests in 155.355s

OK

nosetests tests/foreman/ui/test_syncplan.py -m _delete
.
----------------------------------------------------------------------
Ran 1 test in 154.869s

OK

nosetests tests/foreman/ui/test_trend.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 46.469s

OK

nosetests tests/foreman/ui/test_user.py -m _delete
...
----------------------------------------------------------------------
Ran 3 tests in 135.042s

OK

nosetests tests/foreman/ui/test_usergroup.py -m _delete
..
----------------------------------------------------------------------
Ran 2 tests in 199.938s

OK

nosetests tests/foreman/ui/test_ldapauthsource.py -m _delete
.
----------------------------------------------------------------------
Ran 1 test in 239.965s

OK

nosetests tests/foreman/ui/test_oscapcontent.py -m _delete
.
----------------------------------------------------------------------
Ran 1 test in 291.001s

OK


nosetests tests/foreman/ui/test_docker.py:DockerRegistryTestCase
......
----------------------------------------------------------------------
Ran 6 tests in 504.160s

OK

nosetests tests/foreman/ui/test_environment.py -m test_positive_delete
.
----------------------------------------------------------------------
Ran 1 test in 92.326s

OK

nosetests tests/foreman/ui/test_docker.py:DockerRepositoryTestCase -m _delete
..
----------------------------------------------------------------------
Ran 2 tests in 835.572s

OK
```

### CODE REVIEW

File: robottelo/ui/discoveryrules.py [hunk 1]
```diff
@@ -68,10 +67,6 @@ def search(self, name):
         """Searches existing discovery rule from UI. It is necessary to use
         custom search as we don't have both search bar and search button there.
         """
-        if not bz_bug_is_open(1233135):
```
Comment (svtkachenko) [inline robottelo/ui/discoveryrules.py]: According to BZ, this issue would be tracked via redmine, that's where we can use `rm_bug_is_open` :D

Comment (oshtaier) [inline robottelo/ui/discoveryrules.py]: That one - http://projects.theforeman.org/issues/9021?
so something like `rm_bug_is_open(9021)`?

Comment (svtkachenko) [inline robottelo/ui/discoveryrules.py]: Yes and yes.

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -5007,8 +5007,7 @@ def test_positive_delete_cv_promoted_to_multi_env(self):
             self.content_views.remove_version_from_environments(
                 cv_name, version, list(all_env_names_set))
             # delete content view
-            self.content_views.delete(cv_name)
-            self.assertIsNone(self.content_views.search(cv_name))
```
Comment (svtkachenko) [inline tests/foreman/ui/test_contentview.py]: This assert is intentionally removed?

Comment (oshtaier) [inline tests/foreman/ui/test_contentview.py]: Yes, we don't need assertion after delete operations as it is performed inside of delete method

**Comment by svtkachenko** [comment]
ACK

**Pull request merged**